prefect_gcp/workers/cloud_run_v2.py

```diff
@@ -0,0 +1,838 @@
+import re
+import shlex
+import time
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Optional
+
+from anyio.abc import TaskStatus
+from google.api_core.client_options import ClientOptions
+from googleapiclient import discovery
+
+# noinspection PyProtectedMember
+from googleapiclient.discovery import Resource
+from googleapiclient.errors import HttpError
+from prefect.exceptions import InfrastructureNotFound
+from prefect.logging.loggers import PrefectLogAdapter
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from prefect.utilities.dockerutils import get_prefect_image_name
+from prefect.utilities.pydantic import JsonPatch
+from prefect.workers.base import (
+    BaseJobConfiguration,
+    BaseVariables,
+    BaseWorker,
+    BaseWorkerResult,
+)
+from pydantic import VERSION as PYDANTIC_VERSION
+
+if PYDANTIC_VERSION.startswith("2."):
```

Comment:
```suggestion
from prefect._internal.pydantic import HAS_PYDANTIC_V2

if HAS_PYDANTIC_V2:
```

Reply:
Should I change this in the other blocks as well?

Reply:
ah, good catch. i think we can keep it like this for now until we standardize around our helper check